MLIR.cmake

```diff
@@ -139,7 +142,14 @@ function(add_onnx_mlir_library name)
     )
 
   if (ARG_LINK_LIBS)
-    target_link_libraries(${name} ${ARG_LINK_LIBS})
+    if (ARG_LINK_TARGET_ARCH_LIBS)
```

Comment:
I was thinking that this option would take a list of libraries similar to LINK_LIBS and then map them as opposed to hardcoding all of the options in the function.